src/Lti/Service/GetUserAssignmentLtiRequestService.php

```diff
@@ -124,15 +124,25 @@ public function getAssignmentLtiRequest(Assignment $assignment): LtiRequest
      */
     private function checkIfAssignmentCanBeProcessed(Assignment $assignment): void
     {
-        if (!in_array($assignment->getState(), [Assignment::STATE_READY, Assignment::STATE_STARTED], true)) {
+        $lineItem = $assignment->getLineItem();
+        $maxAttempts = $lineItem->getMaxAttempts();
+        $attemptsCount = $assignment->getAttemptsCount();
+
+        if (
+            $lineItem->hasMaxAttempts() &&
+            (
+                $attemptsCount > $maxAttempts ||
```

Comment:
Is this condition really possible or only for safety?

Reply:
It's a safety net, yes :) Technically it should not be possible.